theano/scalar/basic.py

```diff
@@ -1155,8 +1155,22 @@ def c_code(self, node, name, (x, low, hi), (z, ), sub):
         return ("%(z)s = %(x)s %(cmp1)s %(low)s &&"
                 " %(x)s %(cmp2)s %(hi)s;" % locals())
 
+    def get_grad(self, elem):
+        if elem.type in complex_types:
+            msg = "No gradient implemented for complex numbers in\
+                                 class scalar.basic.InRange"
+            raise NotImplementedError(msg)
+        elif elem.type in discrete_types:
+            return elem.zeros_like().astype(theano.config.floatX)
+        else:
+            return elem.zeros_like()
+
     def grad(self, (x, low, hi), (gz, )):
-        return None, None, None
+        grads = []
+        for elem in [x, low, hi]:
+            grads.append(get_grad(elem))
```

Comment:
pyflakes won't catch anything useful for a while since 90% of the files in theano are whitelisted.

But the problem with the tests is worrying, yes.